oauth2-authorization-server/src/main/java/org/springframework/security/oauth2/server/authorization/oidc/UserInfoClaimsMapper.java

```diff
@@ -0,0 +1,9 @@
+package org.springframework.security.oauth2.server.authorization.oidc;
+
+import org.springframework.security.oauth2.core.oidc.OidcUserInfo;
+
+public interface UserInfoClaimsMapper {
```

Comment:
It's possible we may want to expose something like this in the future. But for now, I think we should reduce visibility to remove this from the public API. See note on `OAuth2AuthorizationServerConfigurer`.

I suggest renaming to `OidcUserInfoClaimsMapper`, moving it to `org.springframework.security.config.annotation.web.configurers.oauth2.server.authorization` (same package as configurer), and changing to the following:

```java
class OidcUserInfoClaimsMapper implements Function<Object, OidcUserInfo> {

	private final Function<Object, OidcUserInfo> delegate;

	OidcUserInfoClaimsMapper(Function<Object, OidcUserInfo> delegate) {
		this.delegate = delegate;
	}

	@Override
	public OidcUserInfo apply(Object principal) {
		return this.delegate.apply(principal);
	}

}
```

Even with reduced visibility, make sure it has appropriate javadoc and copyright.